lib/mpl_toolkits/mplot3d/tests/test_axes3d.py

```diff
@@ -2276,6 +2276,23 @@ def test_on_move_vertical_axis(vertical_axis: str) -> None:
     )
 
 
+@pytest.mark.parametrize("vertical_axis", ["x", "y", "z"])
+def test_set_box_aspect_vertical_axis(vertical_axis: str) -> None:
+    ax = plt.subplot(1, 1, 1, projection="3d")
+    ax.view_init(elev=0, azim=0, roll=0, vertical_axis=vertical_axis)
+    ax.figure.canvas.draw()
+
+    aspect_old = tuple(ax._box_aspect)
+    aspect_expected = np.roll(
+        aspect_old, -1 * (ax._axis_names.index(vertical_axis) - 2)
+    )
+
+    ax.set_box_aspect(None)
+    aspect_new = tuple(ax._box_aspect)
+
+    np.testing.assert_allclose(aspect_expected, aspect_new)
```

Comment:
Does this test have semantic meaning? I can't see by itself that `aspect_expected` is expected to be as defined by the roll operation. To me it appears that this formular was just copied from `set_box_aspect`, which would be tautological ("set_box_aspect behaves according to the formula given in set_box_aspect") and doesn't actually prove anything. Also note that the first two entries of the default box aspect (None -> (4, 4, 3)) are identical. This reduces test power because we cannot distinguish whether they are mixed up or not.

I think there are two ways to make this better: Either

- Use explicit expectations. Something like (not checked the numbers and the exact logic):
  ```
  ax.view_init(elev=0, azim=0, roll=0, vertical_axis="x")
  ax.set_box_aspect((1, 2, 3))  # (x, y, z)
  np.testing.assert_allclose)(ax._box_aspect, (2, 3, 1))  # (width, depth, height)
  ```
  or
- use an image comparison test and compare rotated views with permuted box_aspects.